engine/include/cubos/engine/voxels/grid.hpp

```diff
@@ -91,11 +85,10 @@ namespace cubos::engine
         /// @return Whether the conversion was successful.
         bool convert(const VoxelPalette& src, const VoxelPalette& dst, float minSimilarity);
 
-    private:
-        friend void core::data::old::serialize(core::data::old::Serializer& /*serializer*/, const VoxelGrid& /*grid*/,
-                                               const char* /*name*/);
-        friend void core::data::old::deserialize(core::data::old::Deserializer& /*deserializer*/, VoxelGrid& /*grid*/);
+        bool loadFrom(core::memory::Stream& stream);
+        bool writeTo(core::memory::Stream& stream) const;
```

Comment:
Missing docs. Explain briefly the format used to store/save them here.
I.e.

```suggestion
        /// @brief Loads the grid's data from the given stream.
        ///
        /// Assumes the data is stored in big-endian (network byte order).
        /// The first bytes correspond to three uint32_t, which represent the size of the grid (x, y, z).
        /// The next bytes correspond to `size.x * size.y * size.z` uint16_t, which represent the actual voxel materials.
        /// The voxel data is indexed by `x + y * size.x + z * size.x * size.y`.
        ///
        /// @param stream Stream to read from.
        /// @return Whether the stream contained valid data.
        bool loadFrom(core::memory::Stream& stream);
        
        /// @brief Writs the grid's data to the given stream.
        ///
        /// Writes in the format specified in @ref loadFrom.
        ///
        /// @param stream Stream to write to.
        /// @return Whether the write was successful.
        bool writeTo(core::memory::Stream& stream) const;
```